### PR TITLE
feat: notification worker for bookmark reminder

### PR DESCRIPTION
**Pull request by sshanzel**
Notification worker to call for the bookmark reminder. Awaiting product/design team's response about the email template ID (or if we even want that).

Streams PR: https://github.com/dailydotdev/streams/pull/34

### CODE REVIEW

File: src/workers/notifications/postBookmarkReminder.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { NotificationType } from '../../notifications/common';
+import { NotificationWorker } from './worker';
+import { buildPostContext } from './utils';
+
+interface Data {
+  postId: string;
+  userId: string;
+}
+
+const worker: NotificationWorker = {
```
Comment (capJavert) [inline src/workers/notifications/postBookmarkReminder.ts:12]: Can you use `TypedWorker` please? You can see examples in some recent ones like `commentUpvoteCanceledRep`. If you need any support on it let me know.

Comment (sshanzel) [reply src/workers/notifications/postBookmarkReminder.ts:12]: Oh, cool. Let me check that now 👀

Comment (sshanzel) [reply src/workers/notifications/postBookmarkReminder.ts:12]: Do we have a `TypedWorker` for notifications-related workers? The type doesn't seem to work when applied.

File: src/workers/newNotificationV2Mail.ts [hunk 2]
```diff
@@ -73,6 +73,7 @@ export const notificationToTemplateId: Record<NotificationType, string> = {
   squad_public_submitted: '42',
   squad_public_rejected: '43',
   squad_public_approved: '41',
+  post_bookmark_reminder: '',
```
Comment (sshanzel) [inline src/workers/newNotificationV2Mail.ts:76]: [No equivalent email notification](https://dailydotdev.slack.com/archives/C077EU278D7/p1720706944651969?thread_ts=1720702727.044059&cid=C077EU278D7) as per product/design team.

File: __tests__/workers/notifications.ts [hunk 3]
```diff
@@ -562,6 +563,50 @@ describe('post added notifications', () => {
   });
 });
 
+describe('post bookmark reminder', () => {
+  it('should add notification for the user that set the reminder', async () => {
```
Comment (capJavert) [inline __tests__/workers/notifications.ts:567]: can you add `should be registered` test

**Review by capJavert — APPROVED** [review]

File: __tests__/workers/notifications.ts (hunk 3 above)
Comment (sshanzel) [reply __tests__/workers/notifications.ts:567]: Right. I think I remember that somewhere. I'll add it now.

**Pull request merged**